### PR TITLE
Set a maximum value for the FIM interval synchronization

### PR DESCRIPTION
**Pull request by nicolaspapp**
|Related issue|
|---|
| [43363](https://github.com/wazuh/wazuh/issues/4363)|

## Description
- [x] Increase the interval x2 every failed attempt. When success, reset the initial value (`sync_interval`)

- [x] Add descriptive logs to inform about the remaining time for each sync attempt

- [x] Add a setting `max_sync_interval` to limit the maximum interval timeout. When that value is reached, the interval doesn't grow anymore

- [x] Default value for `max_sync_interval` should be 1 hour

- [x] Add the new setting to the remote configuration 

- [x] Add the new setting to the FIM v2.0 documentation https://github.com/wazuh/wazuh-documentation/pull/2027

Instead of a `sync_max_attemps` key will be adding a new key to establish a maximum synchronization interval:

Option | Default | Description
-- | -- | --
max_sync_interval |   1h  | Maximum allowed synchronization intervals. 

Example configuration
``` xml 
<inventory>
  <sync_interval>5m</sync_interval>
  <max_sync_interval>1h</max_sync_interval>
  <response_timeout>30</response_timeout>
  <sync_queue_size>16384</sync_queue_size>
</inventory>
``` 

## Logs/Alerts example

```
Initializing FIM Integrity Synchronization check. Sync interval is 60 seconds.
FIM Integrity Synchronization check failed. Adjusting sync interval for next run.
Initializing FIM Integrity Synchronization check. Sync interval is 120 seconds.
FIM Integrity Synchronization check failed. Adjusting sync interval for next run.
Initializing FIM Integrity Synchronization check. Sync interval is 240 seconds.
```

## Tests
- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X

### CODE REVIEW

**Comment by nicolaspapp** [comment]
Documentation available on https://github.com/wazuh/wazuh-documentation/pull/2095

File: src/syscheckd/fim_sync.c [hunk 1]
```diff
@@ -26,14 +26,19 @@ static w_queue_t * fim_sync_queue;
...
+    // Keep track of synchronization failures
+    long sync_interval = syscheck.sync_interval;
+
     fim_sync_queue = queue_init(syscheck.sync_queue_size);
 
     while (1) {
-        mdebug2("Performing synchronization check.");
+        bool sync_successful = true;
+        
+        minfo("Initializing FIM Integrity Synchronization check. Sync interval is %li seconds.", sync_interval);
```
Comment (chemamartinez) [inline src/syscheckd/fim_sync.c]: Set this message to debug1

File: src/syscheckd/fim_sync.c [hunk 2]
```diff
@@ -45,7 +50,19 @@ void * fim_run_integrity(void * args) {
...
+
+            sync_successful = false;
+        }
+
+        if (sync_successful) {
+            sync_interval = syscheck.sync_interval;
         }
+        else {
+            // Duplicate for every failure
+            minfo("FIM Integrity Synchronization check failed. Adjusting sync interval for next run.");
```
Comment (chemamartinez) [inline src/syscheckd/fim_sync.c]: Same for this message, it is better to move it to debug1.

**Comment by chemamartinez** [comment]
GJ @nicolaspapp!

**Pull request merged**